File: domain/src/main/java/org/openmedstack/eventstore/ICommitEvents.java
package org.openmedstack.eventstore;

import java.util.concurrent.CompletableFuture;

public interface ICommitEvents {
    CompletableFuture<Iterable<Commit>> getFrom(String bucketId, String streamId, int minRevision, int maxRevision);

    CompletableFuture<Commit> commit(CommitAttempt attempt);
}

File: domain/src/main/kotlin/org/openmedstack/eventstore/ICommitEvents.kt
package org.openmedstack.eventstore

import java.util.concurrent.CompletableFuture

interface ICommitEvents {
    fun getFrom(bucketId: String?, streamId: String, minRevision: Int, maxRevision: Int): CompletableFuture<Iterable<Commit>>
    fun commit(attempt: CommitAttempt): CompletableFuture<Commit>
}
